h2o-py/h2o/frame.py

```diff
@@ -2577,6 +2577,42 @@ def pivot(self, index, column, value):
             raise H2OValueError("'index' argument is not type enum, time or int")
         return H2OFrame._expr(expr=ExprNode("pivot",self,index,column,value))
 
+    def rank_within_group_by(self, groupbyCols, sortCols, ascending=[], newColName="New_Rank_column"):
+        """
+        Return a new frame that is sorted by columns specified in sortCols with sorting direction specified
+        by ascending.  In addition, a new column is added to the frame that denotes the rank of the row
+        within the grouping specified by the groupbyCols.
+
+        :param groupbyCols: The columns to group on (either a single column name/index, or a list of column names
+          or column indices
+        :param sortCols: The columns to sort on (either a single column name/index, or a list of column names or
+          column indices
+        :param ascending: Optional Boolean array to denote sorting direction for each sorting column.  True for
+          ascending, False for descending.  Default is ascending sort.  Sort direction for enums will be ignored.
+        :param newColName: Optional String to denote the new column names.  Default to New_Rank_column.
+
+        :return: a new Frame sorted by columns in sortCols with new rank column within the grouping specified
+          by the groupbyCols.
```

Comment:
Would it be possible to include an example?

Reply:
This is a good idea.  Yes, Nidhi has an example.  Let me asked if she will let me include it.